sdk/java/src/main/java/com/chain/http/Client.java

```diff
@@ -379,19 +379,24 @@ public void setProxy(Proxy proxy) {
   }
 
   private OkHttpClient buildHttpClient(Builder builder) {
-    OkHttpClient httpClient = new OkHttpClient();
+    OkHttpClient httpClient = builder.httpClient.clone();
+    httpClient.setFollowRedirects(false);
 
     if (builder.sslSocketFactory != null) {
       httpClient.setSslSocketFactory(builder.sslSocketFactory);
     }
-
-    httpClient.setFollowRedirects(false);
-    httpClient.setReadTimeout(builder.readTimeout, builder.readTimeoutUnit);
-    httpClient.setWriteTimeout(builder.writeTimeout, builder.writeTimeoutUnit);
-    httpClient.setConnectTimeout(builder.connectTimeout, builder.connectTimeoutUnit);
-
-    httpClient.setConnectionPool(builder.pool);
-
+    if (builder.readTimeoutUnit != null) {
```

Comment:
It feels like if we're instantiating `httpClient` in the builder, these timeouts should be set directly on the builder's `httpClient` object. In other words, maybe `buildHttpClient` is something that moves into the `Builder` class?